step-plans/step-plans-core/src/main/java/step/threadpool/ThreadPool.java

```diff
@@ -135,7 +138,8 @@ public boolean isParallel() {
 	}
 
 	public <WORK_ITEM> void consumeWork(Iterator<WORK_ITEM> workItemIterator,
-			WorkerItemConsumerFactory<WORK_ITEM> workItemConsumerFactory, int numberOfThreads) {
+			WorkerItemConsumerFactory<WORK_ITEM> workItemConsumerFactory, int numberOfThreads,
+										OptionalInt expectedNumberOfThreads) {

```

Comment:
Would rename the parameter as follow:
- numberOfThreads => specifiedNumberOfThreads
- expectedNumberOfThreads => requiredNumberOfThreads

Reply:
done